receiver.go

```diff
@@ -496,13 +501,7 @@ func newReceiver(source string, session *Session, opts *ReceiverOptions) (*Recei
 // attach sends the Attach performative to establish the link with its parent session.
 // this is automatically called by the new*Link constructors.
 func (r *Receiver) attach(ctx context.Context) error {
-	// buffer rx to linkCredit so that conn.mux won't block
-	// attempting to send to a slow reader
-	if r.manualCreditor != nil {
-		r.l.rx = make(chan frames.FrameBody, r.maxCredit)
-	} else {
-		r.l.rx = make(chan frames.FrameBody, r.l.availableCredit)
-	}
+	r.l.rx = make(chan frames.FrameBody, 1)
```

Comment:
Why is this sized so much smaller now?

Reply:
We already size the `messages` channel to that of the link's credit.  Double-buffering seems unnecessary.  It does need to be buffered though so the session mux can hand off the frame to link mux without blocking.

Reply:
Reverted per discussion, opened https://github.com/Azure/go-amqp/issues/217 to track cleaning up the double-buffering.